ArduPlane/mode.h

```diff
@@ -235,6 +237,8 @@ class ModeGuided : public Mode
 
     void update_target_altitude() override;
 
+    bool pre_arm_checks(size_t buflen, char *buffer) const override { return true; }
```

Comment:
This doesn't look right - you're not calling the base class method, which could deny for any number of reasons.

Reply:
Well, there are definitely ways-and-means to get around that.  Notably callbacks like `requires_GPS()` and whatnot to see if a check needs to be done before running in in the base class.

Perhaps just a comment/warning in the base class implementation that not all subclasses call the base class implementaton and that should be fixed before checks added here?

Reply:
I agree that a little call back would be nicer, I just implemented that, costs 120 bytes more, don't want this to get bounced for using too much flash. 

I have added some big comments.